File: src/main/java/io/dustin/finstage/financial/domain/BalanceSheet.java
package io.dustin.finstage.financial.domain;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;

@Getter
@Builder
public class BalanceSheet {
    private final String symbol;
    private final LocalDate date;
    private final long totalAssets;
    private final long totalLiabilities;
    private final long totalEquity;
    private final Long cashAndEquivalents; // nullable
    private final Long inventory;          // nullable

    private BalanceSheet(
            String symbol,
            LocalDate date,
            long totalAssets,
            long totalLiabilities,
            long totalEquity,
            Long cashAndEquivalents,
            Long inventory
    ) {
        this.symbol = symbol;
        this.date = date;
        this.totalAssets = totalAssets;
        this.totalLiabilities = totalLiabilities;
        this.totalEquity = totalEquity;
        this.cashAndEquivalents = cashAndEquivalents;
        this.inventory = inventory;
    }

    public static BalanceSheet of(
            String symbol,
            LocalDate date,
            long totalAssets,
            long totalLiabilities,
            long totalEquity,
            Long cashAndEquivalents,
            Long inventory
    ) {
        return new BalanceSheet(symbol, date, totalAssets, totalLiabilities, totalEquity, cashAndEquivalents, inventory);
    }


}

File: src/main/kotlin/io/dustin/finstage/financial/domain/BalanceSheet.kt
package io.dustin.finstage.financial.domain

import java.time.LocalDate

data class BalanceSheet(
        val symbol: String,
        val date: LocalDate,
        val totalAssets: Long,
        val totalLiabilities: Long,
        val totalEquity: Long,
        val cashAndEquivalents: Long?, // nullable
        val inventory: Long?           // nullable
) {
    companion object {
        fun of(
                symbol: String,
                date: LocalDate,
                totalAssets: Long,
                totalLiabilities: Long,
                totalEquity: Long,
                cashAndEquivalents: Long?,
                inventory: Long?
        ): BalanceSheet {
            return BalanceSheet(
                    symbol,
                    date,
                    totalAssets,
                    totalLiabilities,
                    totalEquity,
                    cashAndEquivalents,
                    inventory
            )
        }
    }
}
